lib/routes.js

```diff
@@ -92,26 +93,38 @@ function fastifySwagger (fastify, opts, done) {
     }
   })
 
+  const hasTransformSwaggerFn = typeof opts.transformSwagger === 'function'
+  const transformSwagger = opts.transformSwagger
   fastify.route({
     url: '/json',
     method: 'GET',
     schema: { hide: true },
     ...hooks,
-    handler: function (req, reply) {
-      reply.send(fastify.swagger())
-    }
+    handler: hasTransformSwaggerFn
+      ? function (req, reply) {
+        reply.send(transformSwagger(req, reply, JSON.parse(JSON.stringify(fastify.swagger()))))
```

Comment:
I do JSON.parse(JSON.stringify)) to get an immutable object. I dont think that this is smart.

Reply:
By "immutable" do you mean a clone/copy that can be mutated for a single request?

Reply:
I suggest to use https://github.com/davidmarkclements/rfdc

Reply:
please see my comment https://github.com/fastify/fastify-swagger-ui/pull/20#issuecomment-1283668663

Reply:
using rfdc instead of stringify parse